src/components/CounterInput/index.js

```diff
@@ -0,0 +1,245 @@
+        tabIndex,
+        onFocus,
+        onBlur,
+        onClick,
+        onKeyDown,
+        disabled,
+        readOnly,
+        required,
+        min,
+        max,
+        isBare,
+        error,
+        variant,
+        step,
+        label,
+        labelAlignment,
+        hideLabel,
+        bottomHelpText,
+        className,
+        style,
+    } = useReduxForm(props);
+
+    const inputRef = useRef(null);
+    const inputId = useUniqueIdentifier('counter-input');
+    const errorMessageId = useErrorMessageId(error);
+    const labelId = useLabelId(label);
+    const isReadOnly = !!(!disabled && readOnly);
+
+    useImperativeHandle(ref, () => ({
+        focus: () => {
+            inputRef.current.focus();
+        },
+        click: () => {
+            inputRef.current.click();
+        },
+        blur: () => {
+            inputRef.current.blur();
+        },
+    }));
+
+    const handlePlusMouseDown = event => {
+        event.preventDefault();
+        inputRef.current.focus();
+        onChange(getNormalizedValue(getValue(Number(value)) + step));
+    };
+
+    const handleMinusMouseDown = event => {
+        event.preventDefault();
+        inputRef.current.focus();
+        onChange(getNormalizedValue(getValue(Number(value)) - step));
+    };
+
+    const handleEvents = (event, callBack) => {
+        if (event.target.value === '') {
+            return callBack();
+        }
+        return callBack(Number(event.target.value));
+    };
+
+    return (
+        <StyledContainer id={id} className={className} style={style}>
+            <Label
+                label={label}
+                labelAlignment={labelAlignment}
+                hideLabel={hideLabel}
+                required={required}
+                inputId={inputId}
+                readOnly={isReadOnly}
+                id={labelId}
+            />
+            <RelativeElement>
+                <ButtonContainer iconPosition="left" readOnly={readOnly} error={error}>
+                    <StyledButton
+                        variant="base"
+                        size="small"
+                        icon={<MinusIcon />}
+                        onMouseDown={handleMinusMouseDown}
+                        tabIndex={-1}
+                        disabled={isButtonDisabled(isMin(value, step, min), disabled, readOnly)}
+                        assistiveText="minus button"
+                    />
+                </ButtonContainer>
+                <StyledInput
+                    id={inputId}
+                    name={name}
+                    type="number"
+                    value={value}
+                    placeholder={placeholder}
+                    onChange={event => handleEvents(event, onChange)}
+                    tabIndex={tabIndex}
+                    onFocus={event => handleEvents(event, onFocus)}
+                    onBlur={event => handleEvents(event, onBlur)}
+                    onClick={event => handleEvents(event, onClick)}
```

Comment:
let pass onClick direct, onFocus, and onBlur are the ones that need to be equal onChange